network/network_node_repository.go

```diff
@@ -413,32 +414,48 @@ func addWorksForUid(originalNode, inputNode *models.NetworkNode) error {
 	return nil
 }
 
-func GetNetworkNodeByCode(code string) *models.NetworkNode {
+var ErrNodeNotFound = errors.New("node not found")
+
+func GetNetworkNodeByCode(code string) (*models.NetworkNode, error) {
 	var node *models.NetworkNode
 
 	if code == "" {
 		log.Println("[GetNetworkNodeByCode] code empty")
-		return nil
+		return nil, fmt.Errorf("empty code")
 	}
 
 	iter := lib.WhereFirestore(lib.NetworkNodesCollection, "code", "==", code)
 	nodeDocSnapshot, err := iter.Next()
 
 	if err == iterator.Done && nodeDocSnapshot == nil {
 		log.Println("[GetNetworkNodeByCode] node not found")
-		return nil
+		return nil, ErrNodeNotFound
 	}
 
 	if err != iterator.Done && err != nil {
 		log.Printf("[GetNetworkNodeByCode] error getting node: %s", err.Error())
-		return nil
+		return nil, err
 	}
 
 	err = nodeDocSnapshot.DataTo(&node)
 	if node == nil || err != nil {
 		log.Printf("[GetNetworkNodeByCode] could not parse node: %s", err)
-		return nil
+		return nil, err
 	}
 
-	return node
+	return node, nil
+}
+
+func TestNetworkNodeUniqueness(nodeCode string) error {
+	_, err := GetNetworkNodeByCode(nodeCode)
+	// If err == nil, node found
+	if err == nil {
+		return errors.New("node already exists")
+	}
+	if !errors.Is(err, ErrNodeNotFound) {
+		// If err is *not* ErrNodeNotFound, an error raised hitting the db
+		log.Printf("error checking node code: %s", err.Error())
```

Comment:
Let's try to enforce a standard here (if @Yousef-Hammar agrees):
_whenever we can, let's leave the logs to the caller_ -> in this case it will be the two API handlers.